pandas/_libs/tslibs/offsets.pyx

```diff
@@ -314,6 +314,9 @@ cdef _determine_offset(kwds):
 
     use_relativedelta = False
     if len(kwds_no_nanos) > 0:
+        if "milliseconds" in kwds_no_nanos:
+            return timedelta(**kwds_no_nanos), use_relativedelta
```

Comment:
Can you assign to `offset` and `use_relativedelta` and just let the function return naturally? I think flows better with the other branching